fileio/fileio.go

```diff
@@ -105,10 +105,29 @@ func PeekReal(reader *bufio.Reader, n int) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	} else {
-		return reader.Peek(n)
+		return peek, err
 	}
 }
 
+// ReadHeader will advance a reader past initial lines that begin with '#',
+// returning a slice of these comments lines and leaving the reader at
+// the first non-comment line
+func ReadHeader(reader *bufio.Reader) ([]string, error) {
+	var peek []byte
+	var peekErr error
+	var header []string
+	var line string
+	for peek, peekErr = reader.Peek(1); peekErr == nil && peek[0] == '#'; peek, peekErr = reader.Peek(1) {
+		line, _ = NextLine(reader)
+		header = append(header, line)
+	}
+
+	if peekErr == io.EOF {
+		return header, nil
+	}
+	return header, peekErr
```

Comment:
Correct.  There are three cases: 1) The file is only the header, and in this case the header ends when we get peekErr equals io.EOF  (this returns on 126).  2) We read the whole header and realize the next line is a data line because peek[0] != '#'  (this returns on 128).  3) We get an error (that is not io.EOF) while reading the header and this also returns on line 128.